### PR TITLE
e2e-tests: headlamp.spec: Add test for pagination

### PR DESCRIPTION
**Pull request by illume**
So we can see it updates which page it is on.

### CODE REVIEW

File: e2e-tests/tests/headlamp.spec.ts [hunk 1]
```diff
@@ -70,20 +70,46 @@ test('Service account tab should have headlamp-admin', async ({ page }) => {
...
+  );
 });
+
+test("pagination goes to next page", async ({ page }) => {
+  const headlampPage = new HeadlampPage(page);
+  await headlampPage.authenticate();
+
+  await page.click('span:has-text("Security")');
+  await page.waitForLoadState("load");
+  await page.waitForSelector('span:has-text("Service Accounts")');
```
Comment (knrt10) [inline e2e-tests/tests/headlamp.spec.ts:99]: We could use existing method on class 
```suggestion

  const securityPage = new SecurityPage(page);
  await securityPage.navigateToSecurity();
  await securityPage.clickOnServiceAccountsSection();
```

File: e2e-tests/tests/headlamp.spec.ts [hunk 2]
```diff
@@ -70,20 +70,46 @@ test('Service account tab should have headlamp-admin', async ({ page }) => {
...
+
+  await page.click('span:has-text("Roles")');
+  await page.waitForSelector('span:has-text("Roles")');
+  await page.waitForSelector('p:has-text("Rows per page")');
```
Comment (knrt10) [inline e2e-tests/tests/headlamp.spec.ts:103]: We can create a new method `clickOnRolesSection` and use it here for future use.

File: e2e-tests/tests/headlamp.spec.ts [hunk 3]
```diff
@@ -70,20 +70,46 @@ test('Service account tab should have headlamp-admin', async ({ page }) => {
...
+
+  const paginationCaption = page.locator("p:has-text(' of ')");
+  const captionText = await paginationCaption.textContent();
+
+  const nextPageButton = page.getByTitle("Next page");
+  await nextPageButton.click();
+  const captionText2 = await paginationCaption.textContent();
+
+  expect(captionText).not.toBe(captionText2);
```
Comment (knrt10) [inline e2e-tests/tests/headlamp.spec.ts:112]: This can be moved to new method like `checkPageCaption`

**Review by knrt10 — COMMENTED** [review]
Some suggestions. Thanks for adding these

**Comment by vyncent-t** [comment]
reviewing w/ extension to test changes

**Comment by vyncent-t** [comment]
> reviewing w/ extension to test changes

mostly running into the timeout error, i have tried different env sets and so far only the first test of ` GET /plugins/list returns plugins list` succeeds, the rest timeout for me in extension mode and manual test running but it could just be an error on my end

image

**Comment by illume** [comment]
I'm not sure why putting these things into methods made it not pass in CI. 

- old commit 8d6ca506b038c87d64d4b4c320066d560237113c 
- new commit 2acb53d2b674c674b60332621723ec092865221e

**Comment by knrt10** [comment]
I will update the PR. Thank you for the work @illume

**Comment by knrt10** [comment]
@illume I have updated the PR, PTAL. Thanks

**Comment by illume** [comment]
Thanks @knrt10

**Pull request merged**